torch/csrc/autograd/custom_function.h

```diff
@@ -41,11 +56,37 @@ struct CFunction {
   static variable_list apply(Args&&... args);
 };
 
+
+// Context to save information during forward that can be accessed in backward
 struct AutogradContext {
-  std::vector<torch::autograd::SavedVariable> saved_variables;
+  // Can be used to save non-variable data for backward()
+  std::unordered_map<std::string, at::IValue> save;
```

Comment:
Not sure about this name. Maybe "user_data"? I'm really not sure if it's better.